toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/utils/ConfigTesterHelper.java

```diff
@@ -0,0 +1,221 @@
+
+import io.toast.tk.agent.ui.NotificationManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * Configuration panel
+ */
+public class ConfigTesterHelper {
+
+	private static final String FILE_DIRECTORY_SEPARATOR = "/";
+	private static final int timeout = 500;
+	
+	public static boolean testWebAppDirectory(String directory, boolean runTryValue, boolean fileOrDirectory)
+			throws IOException {
+		String fileName = directory.split(FILE_DIRECTORY_SEPARATOR)[directory.split(FILE_DIRECTORY_SEPARATOR).length - 1];
+		
+		boolean isKo = true;
+		if (fileOrDirectory) {
+			if (testFileDirectory(directory, runTryValue, fileName)) {
+				isKo = false;
+			}
+		} else {
+			if (testDirectory(directory, runTryValue, fileName)) {
+				isKo = false;
+			}
+		}
+		return !isKo;
+	}
+
+	public static boolean testDirectory(String directory, boolean runTryValue, String fileName) {
+		File myFile = new File(directory);
+		if (myFile.exists()) {
+			if (myFile.isDirectory()) {
+				return true;
+			} else {
+				if (runTryValue) {
+					NotificationManager.showMessage("You did not select a directory.").showNotification();
+				}
+				return false;
+			}
+		} else {
+			if (runTryValue) {
+				NotificationManager.showMessage("The directory : " + directory.split(fileName)[0] + " does not exist !")
+						.showNotification();
+			}
+			return false;
+		}
+	}
+
+	public static boolean testFileDirectory(String directory, boolean runTryValue, String fileName) {
+		File myFile = new File(directory);
+		if (myFile.exists()) {
+			if (myFile.isFile()) {
+				return true;
+			} else {
+				if (runTryValue) {
+					NotificationManager.showMessage("You did not select a file.").showNotification();
+				}
+				return false;
+			}
+		} else {
+			if (runTryValue) {
+				NotificationManager
+						.showMessage(fileName + " does not exist in the directory : " + directory.split(fileName)[0])
+						.showNotification();
+			}
+			return false;
+		}
+	}
+
+	public static boolean testWebAppUrl(String url, boolean runTryValue) throws IOException {
+		return testWebAppUrl(url, runTryValue, null, null, null, null);					
+	}
+
+	public static boolean testWebAppUrl(String url, boolean runTryValue, String proxyAdress, String proxyPort,
+			String proxyUserName, String proxyUserPswd) throws IOException {
+
+		boolean isKo = true;
+		if (url.contains(" ")) {
+			if (runTryValue) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)